OpenRA.Mods.Common/Activities/Air/Fly.cs

```diff
@@ -57,6 +57,9 @@ public static void FlyToward(Actor self, Aircraft plane, int desiredFacing, WDis
 
 		public override Activity Tick(Actor self)
 		{
+			// Refuse to take off if it would land immediately again.
+			if (plane.ForceLanding) return null;
```

Comment:
Activity is explicitly doing this when keepQueue is false, this means it will return null only if the current activity is supposed to cancel all future activities forever until a new one is added.